src/utilities/externalEvents.ts

```diff
@@ -0,0 +1,47 @@
+import type { ExternalEvent, ExternalEventId, ExternalEventPkey, ExternalEventType } from '../types/external-event';
+import type {
+  DerivationGroup,
+  ExternalSourcePkey,
+  ExternalSourceSlim,
+  ExternalSourceType,
+} from '../types/external-source';
+
+// External Event Row/Hash Functions
+export function getRowIdExternalEventWhole(externalEvent: ExternalEvent): ExternalEventId {
+  return (
+    externalEvent.pkey.derivation_group_name +
+    externalEvent.pkey.source_key +
+    externalEvent.pkey.event_type_name +
+    externalEvent.pkey.key
+  );
+}
+
+export function getRowIdExternalEvent(externalEventPkey: ExternalEventPkey): ExternalEventId {
+  return (
+    externalEventPkey.derivation_group_name +
+    externalEventPkey.source_key +
+    externalEventPkey.event_type_name +
+    externalEventPkey.key
+  );
+}
```

Comment:
Reason for having both here?

Reply:
There are a few places such as `stores/external-event.ts` that pass the full `ExternalEvent` object vs. just the `pkey` field. Would it be better to overload the `getRowIdExternalEvent` function, or figure out how to just always pass the `pkey`? I was having trouble especially with the example from `stores/external-event.ts` as it's used in a call to lodash's `keyBy`

Reply:
I think that the issue is that the logic for creating the ID is being duplicated, which makes maintenance a little bit harder (not much, but just something we have to be careful of now). If it can be reduced to just always passing in the `pkey` or always using the whole event, then that would be the most ideal. The next ideal solution would be to have `getRowIdExternalEventWhole` call `getRowIdExternalEvent` directly. That way `getRowIdExternalEvent` is the only function where the logic for combining ids exists.